overlord/devicestate/devicemgr.go

```diff
@@ -177,6 +176,15 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, runner *state.T
 	// wire FDE kernel hook support into boot
 	boot.HasFDESetupHook = m.hasFDESetupHook
 	boot.RunFDESetupHook = m.runFDESetupHook
+	// The fde-setup "device-setup" is called during install in which
+	// the state is unlocked
+	// XXX: instead of passing a "fde.SetupRequest" pass just []byte?
+	//      see https://github.com/snapcore/snapd/compare/master...mvo5:ice/refactor-fde?expand=1
+	boot.RunFDEDeviceSetupHook = func(req *fde.SetupRequest) ([]byte, error) {
```

Comment:
do we need two functions?  we could check the op value instead?  (that in itself is not great tbh but I think it's a bit better that exposing details of devicemgr to gadget/install)